libs/conversion/src/lib/conversion-browse-files-modal/conversion-browse-files-modal.component.ts

```diff
@@ -0,0 +1,173 @@
+  templateUrl: './conversion-browse-files-modal.component.html',
+  styleUrls: ['./conversion-browse-files-modal.component.less']
+})
+export class ConversionBrowseFilesModalComponent extends BrowseFilesModalComponent implements OnInit {
+  private _format: string;
+  formats;
+  @Input() files: ExtendedFileModel[];
+  @Output() selectAll = new EventEmitter<boolean>();
+
+  dynamicOptions: Option[] = [];
+
+  constructor(_uploadService: UploadFilesService,
+    private _conversionService: ConversionService,
+    private _modalService: ModalService) {
+    super(_uploadService);
+  }
+
+  selectAllItems(checked: boolean){
+    this.selectAll.emit(checked);
+
+    this.dynamicOptions = this.prepareMultipleConversionTypes();
+  }
+
+  selectSingleItem(checked: boolean, file: ExtendedFileModel){
+    // TODO: refactor?
+    const selectedFiles = this.files.filter(f => f.guid === file.guid);
+    if (selectedFiles.length === 1){
+      selectedFiles[0].selected = checked;
+    }
+
+    this.dynamicOptions = this.prepareMultipleConversionTypes();
+  }
+
+  getLabelString(){
+    if (this.files && this.files.length > 0) {
+      const selectedCount = this.files.filter(file => file.selected).length;
+      if (selectedCount > 0) {
+      return 'Add ' + selectedCount + ' selected'
+      }
+      else {
+        return 'Add selected'
+      }
+    }
+  }
+
+  prepareMultipleConversionTypes() {
+    const allTypes = [];
+
+    this.files.filter(file => file.selected).forEach((f) => {
+      if (f.conversionTypes.length > 0) {
+        const types = Object.assign([], f.conversionTypes);
+        allTypes.push(types);
+      }
+    });
+
+    let longestArray = [];
+    allTypes.forEach((item) => {
+      if(item.length >= longestArray.length){
+        longestArray = item;
+      }
+    });
+
+    //add warnings
+    allTypes.forEach(typesArr => {
+        for (let i = 0; i < longestArray.length; i++) {
+          const type = (longestArray[i].value) ? longestArray[i].value : longestArray[i];
+          // TODO: reconsider second check
+          if (typesArr.indexOf(type) === -1 && typesArr.filter(t => t.name === type).length === 0) {
+              longestArray[i] = {
+                value: type,
+                name: type,
+                warning: true,
+                icon: this.getFormatIcon({name: 'dummyName.' + type, directory: false} as FileModel)
+              };
+          } else {
+            if(!longestArray[i].warning) {
+              longestArray[i] = {
+                value: type,
+                name: type,
+                warning: false,
+                icon: this.getFormatIcon({name: 'dummyName.' + type, directory: false} as FileModel)
+              };
+            }
+          }
+        }
+    }); 
+
+    return longestArray;
+  }
+
+  selectFormat($event: any, file: ExtendedFileModel) {
+    // the case when we selecting format inline on the single
+    if (file) file.selected = true;
```

Comment:
If format for single file was selected after selecting multiple files only that single file should be converted into conversion item.